conda_forge_tick/migrators/version.py

```diff
@@ -492,7 +493,69 @@ def migrate(
             attrs["new_version_errors"][version] = _fmt_error_message(errors, version)
             return {}
 
-        if len(list(_gen_key_selector(cmeta.meta, "source"))) > 0:
+        def _try_simple_update():
```

Comment:
Let's make this a full function with inputs etc outside of here. There is a lot of logic and separate tests of it might be useful.